### ISSUE TITLE
Verify hydrogen on flathub^

### ISSUE THREAD

**Issue by goebbe** [hydrogen-music/hydrogen#1974]
Hydrogen version *          : stabel flathub version
Operating system + version  :  linux flatpak

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Flathub provides an "**unverfied**" version of Hydrogen, therefore the user has
to verify himself, if the package is build from a trusted source.

To see this, you can follow the link to Hydrogen on flathub:
https://flathub.org/apps/org.hydrogenmusic.Hydrogen
A yellow sign "unverified" is displayed. 

Installing Hydrogen from Flathub would be more convenient, if the package could
be marked as "verified" - which would provide some evidence that the package is 
published directly by its maintainers (or by a trusted person).

Here is the documentation about verification: 
https://docs.flathub.org/docs/for-app-authors/verification

**Comment by theGreatWhiteShark** [comment]
Hey @goebbe ,

We actually discourage people to use the Flathub version of Hydrogen at all. It's JACK support is broken (see https://github.com/flathub/org.hydrogenmusic.Hydrogen/issues/33) and we have no way of fixing it.

Instead, we provide AppImages for all Linux folks.

**Comment by goebbe** [comment]
@theGreatWhiteShark Thank for your comment!
In this case, there should be a comment on the description of Hydrogen (on flathub). Something like: 
"This package does not support JACK sound-server directly. See Hydrogen webpage for requirements and recommended means of installation. " 

I would nevertheless recommend keeping a **verified** flatpak available for users of pipewire / pipewire-jack, which seems to work o.k. on some distributions.

Of course, if flatpak is not supported at all, it would probably be best to remove the Hydorgen package from flathub - and discourage its use on the Hydrogen webpage. (I din't find the info on the webpage, but perhaps I just missed it)

**Comment by theGreatWhiteShark** [comment]
> For my use case "i.e. playing around, experimenting, discover new things, having fun", I would probably, not even consider installing "JACK" (if not strictly required). But of course I understand that this may not be the intended purpose and ambition of Hydrogen.

Well, it is our default driver for Linux and it is installed very quickly via copy-paste as many tutorials suggest to use it. The Flathub version of Hydrogen actually came to my attention through a bug report in forum where an user struggled with this very bug.

In general, the best solution would be for Hydrogen to dynamically check whether JACK or pipewire is installed on the system. But this requires a deeper understanding of flatpak itself and at this point I'm not willing to invest significant amounts of time in this direction. I already spent at least two weeks and it ended quite frustrating and with the use of AppImage instead.

> Nevertheless, I would suggest keeping a verified flatpak available - for users of pipewire / pipewire-jack, which seems to work o.k. on some distributions.
>
> If flatpak is not supported at all, it would probably be best to remove the Hydorgen package from flathub - and discourage its use on the Hydrogen webpage. (I din't find the info on the webpage, but perhaps I just missed it)

The package on Flathub is maintained by a third party and we do not have control over the patches applied. On the other hand, this is free software and everyone is allowed to distribute it. And let's not forget that this person put work and time into getting Hydrogen on Flathub! It's an unfortunate state that it does not work properly but at some point I decided to let things go, provide an official AppImage as an alternative, and discourage the usage on issue/forum level instead of putting a disclaimer on our web page.

> In this case, I recommend adding a comment on the description of Hydrogen (on flathub). Something like:
"This package does not support JACK sound-server directly. See Hydrogen webpage for recommended means of installation. "

Agreed. Could you open a ticket at the corresponding [repo](https://github.com/flathub/org.hydrogenmusic.Hydrogen)?

**Comment by goebbe** [comment]
Done! Thank you for your answers and explanations!

**Comment by theGreatWhiteShark** [comment]
> Done!

Shall we close the issue?

**Comment by theGreatWhiteShark** [comment]
> The issue on flathub has been closed - as the developers thinks that this is only an issue of the underlying Linux distributions. He does not consider pointing the users to installing pipewire and pipewire-JACK or to point to the upstream docs.

Yeah. This is what I experienced too and which is the main thing that keeps me from verifying it. Good ol' systemd approach to blame deficiencies in ones client library/application on upstream distributions.

But thanks for giving it a try!
